### PR TITLE
Handling use, mkfs_options, fstopt and mountby elements

### PR DESCRIPTION
**Pull request by imobachgs**
It should fix [bsc#1061042](https://bugzilla.suse.com/show_bug.cgi?id=1061042).

### CODE REVIEW

File: src/lib/y2storage/proposal/autoinst_space_maker.rb [hunk 1]
```diff
@@ -72,11 +72,27 @@ def delete_stuff(devicegraph, disk, drive_spec)
...
+      #
+      # @param devicegraph [Devicegraph]
+      # @param disk        [Disk]
+      # @param use         [String,Array<Integer>] Use value ("all", "linux", "free"
+      #   or a list of partition numbers)
+      def delete_by_use(devicegraph, disk, use)
+        return if use == "free"
+        case use
         when "all"
           disk.partition_table.remove_descendants if disk.partition_table
```
Comment (joseivanlopez) [inline src/lib/y2storage/proposal/autoinst_space_maker.rb]: Any reason to not delete the partition table?

File: test/y2storage/proposal/autoinst_devices_planner_test.rb [hunk 2]
```diff
@@ -187,6 +191,26 @@
         expect(root.filesystem_type).to eq(Y2Storage::Filesystems::Type::EXT4)
         expect(home.filesystem_type).to eq(Y2Storage::Filesystems::Type::XFS)
       end
+
+      it "sets the mountby properties" do
```
Comment (joseivanlopez) [inline test/y2storage/proposal/autoinst_devices_planner_test.rb]: Here the filesystem is checked, but not the mountby, right?

File: src/lib/y2storage/proposal/autoinst_space_maker.rb (hunk 1 above)
Comment (imobachgs) [reply src/lib/y2storage/proposal/autoinst_space_maker.rb]: That's a fair question. From the AutoYaST point of view, these are different use cases. If you want to wipe the partition table, you set the `initialize` element in the profile; if you only want to use the whole disk (but keeping the partition table if it exists), you set `use=all` (this case).

File: test/y2storage/proposal/autoinst_devices_planner_test.rb (hunk 2 above)
Comment (imobachgs) [reply test/y2storage/proposal/autoinst_devices_planner_test.rb]: Yes, you are absolutely right. I will fix it.

File: src/lib/y2storage/proposal/autoinst_space_maker.rb (hunk 1 above)
Comment (joseivanlopez) [reply src/lib/y2storage/proposal/autoinst_space_maker.rb]: Ok, then your code does what it is expected :). Thanks.

File: src/lib/y2storage/storage_manager.rb [hunk 3]
```diff
@@ -190,12 +190,13 @@ def staging_changed?
...
     #
     # Beware: this method can cause data loss
-    def commit
+    def commit(force_rw: false)
       # Tell FsSnapshot whether Snapper should be configured later
       Yast2::FsSnapshot.configure_on_install = configure_snapper?
       callbacks = Callbacks::Commit.new
       storage.calculate_actiongraph
-      storage.commit(callbacks)
+      commit_options = ::Storage::CommitOptions.new(force_rw)
```
Comment (joseivanlopez) [inline src/lib/y2storage/storage_manager.rb]: Maybe `CommitOptions` should have its own wrapper class.

Comment (imobachgs) [reply src/lib/y2storage/storage_manager.rb]: As we do not need any special behavior (it is just some kind of 'struct' to pass options), I would keep it simple by now.

**Pull request merged**